code/modules/mob/say.dm

```diff
@@ -99,6 +99,19 @@
 ///The amount of items we are looking for in the message
 #define MESSAGE_MODS_LENGTH 6
 
+/mob/proc/check_for_custom_say_emote(message, list/mods)
+	var/customsaypos = findtext(message, "*")
+	if(!customsaypos)
+		return message
+	if (is_banned_from(ckey, "Emote"))
+		return copytext(message, customsaypos + 1)
+	mods[MODE_CUSTOM_SAY_EMOTE] = lowertext(copytext_char(message, 1, customsaypos))
+	message = copytext(message, customsaypos + 1)
+	if (!message)
+		mods[MODE_CUSTOM_SAY_ERASE_INPUT] = TRUE
+		message = "an interesting thing to say"
```

Comment:
?

Reply:
It's hard to explain but this basically converts messages that only contain a custom emote, to an emote, IIRC